File: src/plugin/combat/magic/ancient/teleport/KharyrllTeleportSpell.java
package plugin.combat.magic.ancient.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class KharyrllTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 66;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(LAW_RUNE, 2, BLOOD_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(3492, 3471, 0);
	}
	
	@Override
	public int spellId() {
		return 42;
	}
	
	@Override
	public double exp() {
		return 76;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
}

File: src/main/java/plugin/interaction/combat/magic/ancient/teleport/KharyrllTeleportSpell.kt
package plugin.interaction.combat.magic.ancient.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
class KharyrllTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 66
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.LAW_RUNE, 2, MagicConstants.BLOOD_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(3492, 3471, 0)
    }

    override fun spellId(): Int {
        return 42
    }

    override fun exp(): Double {
        return 76.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }
}